### PR TITLE
upgrade to go1.11

### PR DESCRIPTION
**Pull request by BenTheElder**
/assign @ixdy

### CODE REVIEW

**Comment by BenTheElder** [comment]
after https://github.com/kubernetes/test-infra/pull/9475 I'll bump the experimental image to go1.11 as well

File: WORKSPACE [hunk 1]
```diff
@@ -8,18 +8,20 @@ load("@bazel_skylib//:lib.bzl", "versions")
 
 versions.check(minimum_bazel_version = "0.10.0")
 
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
```
[the remark this answers is not in the document]
Comment (ixdy) [reply WORKSPACE]: also apparently `new_http_archive` is also deprecated.
https://docs.bazel.build/versions/master/be/workspace.html#new_http_archive

[the remark this answers is not in the document]
Comment (ixdy) [reply WORKSPACE]: and `http_file`. it just keeps going!

File: WORKSPACE [hunk 2]
```diff
@@ -8,18 +8,20 @@ load("@bazel_skylib//:lib.bzl", "versions")
...
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "1868ff68d6079e31b2f09b828b58d62e57ca8e9636edff699247c9108518570b",
-    url = "https://github.com/bazelbuild/rules_go/releases/download/0.11.1/rules_go-0.11.1.tar.gz",
+    sha256 = "97cf62bdef33519412167fd1e4b0810a318a7c234f5f8dc4f53e2da86241c492",
+    urls = ["https://github.com/bazelbuild/rules_go/releases/download/0.15.3/rules_go-0.15.3.tar.gz"],
 )
 
-load("@io_bazel_rules_go//go:def.bzl", "go_rules_dependencies", "go_register_toolchains")
+load("@io_bazel_rules_go//go:def.bzl", "go_rules_dependencies", "go_register_toolchains", "go_context")
```
Comment (ixdy) [inline WORKSPACE]: do we use `go_context`?

File: testgrid/BUILD.bazel [hunk 3]
```diff
@@ -50,3 +51,26 @@ filegroup(
     tags = ["automanaged"],
     visibility = ["//visibility:public"],
 )
+
+proto_library(
```
Comment (ixdy) [inline testgrid/BUILD.bazel]: I'm not sure whether these changes are right.

**Comment by ixdy** [comment]
some of this could be squashed, but eh.

/lgtm

**Pull request merged**